src/camera/camera-orbit-control.ts

```diff
@@ -1,94 +1,321 @@
-import { ObservablePoint } from "@pixi/math"
 import type { InteractionEvent } from "@pixi/interaction"
-import { Camera } from "./camera"
+import { ObservablePoint } from "@pixi/math"
+import { Compatibility } from "../compatibility/compatibility"
 import { Quat } from "../math/quat"
 import { Vec3 } from "../math/vec3"
-import { Compatibility } from "../compatibility/compatibility"
+import { Camera } from "./camera"
 
 /**
  * Allows the user to control the camera by orbiting the target.
  */
 export class CameraOrbitControl {
-  private _grabbed = false
+  protected _autoUpdate = true;
 
-  private _dampingAngles = new ObservablePoint(() => { }, undefined, 0, 180)
+  /** 
+   * Whether to auto update the camera on prerender.
+   * Default is true.
+   */
+  get autoUpdate(): boolean {
+    return this._autoUpdate
+  }
 
-  private _distance = 5
-  private _dampingDistance = 5
+  set autoUpdate(value: boolean) {
+    this._autoUpdate = value
+  }
+
+  protected _allowControl = true
+
+  /** 
+   * Allows the camera to be controlled by user.
+   */
+  get allowControl(): boolean {
+    return this._allowControl
+  }
+
+  set allowControl(value: boolean) {
+    this._allowControl = value
+  }
 
-  private _angles = new ObservablePoint(() => {
-    this._angles.x = Math.min(Math.max(-85, this._angles.x), 85)
-  }, undefined, 0, 180)
+  protected _camera = Camera.main
+
+  /** 
+   * The camera being controlled.
+   */
+  get camera(): Camera {
+    return this._camera
+  }
+
+  set camera(value: Camera) {
+    this._camera = value
+  }
+
+  protected _target = { x: 0, y: 0, z: 0 }
+
+  /** 
+   * Target position (x, y, z) to orbit.
+   */
+  get target(): { x: number; y: number; z: number } {
+    return this._target
+  }
+
+  set target(value: { x: number; y: number; z: number }) {
+    this._target = value
+  }
+
+  protected _orbit = { x: 0, y: 180 }
+
+  private _angles = new ObservablePoint(
+    () => {
+      this._orbit.x = Math.min(Math.max(-85, this._orbit.x), 85)
```

Comment:
This is now changing the _orbit, but _orbit is never read anywhere?

Reply:
I think this can be exactly what is was previously, before your changes or did I miss something?

Reply:
Ah yes. I originally changed this because _angles referencing itself was causing infinite recursion on mobile. I'll fix the backing property properly.

Reply:
K this should be fixed properly now.